sentry-android-core/src/test/java/io/sentry/android/core/internal/gestures/SentryGestureListenerClickTest.kt

```diff
@@ -0,0 +1,242 @@
+        val context: Context = mock()
+        val resources: Resources = mock()
+        val options = SentryAndroidOptions().apply {
+            dsn = "https://key@sentry.io/proj"
+        }
+        val hub: IHub = mock()
+
+        fun getSut(
+            target: View,
+            resourceName: String = "test_button",
+        ): SentryGestureListener {
+            resources.mockForTarget(target, resourceName)
+            whenever(context.resources).thenReturn(resources)
+            whenever(target.context).thenReturn(context)
+            return SentryGestureListener(
+                WeakReference(window),
+                hub,
+                options,
+                true
+            )
+        }
+    }
+
+    private val fixture = Fixture()
+
+    @Test
+    fun `when target and its ViewGroup are clickable, captures a breadcrumb for target`() {
+        val event: MotionEvent = mock()
+        val container1: ViewGroup = mockView(
+            id = View.generateViewId(),
+            event = event,
+            touchWithinBounds = false,
+        )
+        val target: View = mockView(
+            id = View.generateViewId(),
+            event = event,
+            clickable = true
+        )
+        val notClickableInvalidTarget: View = mockView(
+            id = View.generateViewId(),
+            event = event,
+        )
+        val notVisibleInvalidTarget: View = mockView(
+            id = View.generateViewId(),
+            event = event,
+            visible = false,
+        )
+        val container2: ViewGroup = mockView(
+            id = View.generateViewId(),
+            event = event,
+            clickable = true
+        ) {
+            whenever(it.childCount).thenReturn(3)
+            whenever(it.getChildAt(0)).thenReturn(notClickableInvalidTarget)
+            whenever(it.getChildAt(1)).thenReturn(notVisibleInvalidTarget)
+            whenever(it.getChildAt(2)).thenReturn(target)
+        }
+        fixture.window.mockDecorView<ViewGroup>(
+            event = event,
+        ) {
+            whenever(it.childCount).thenReturn(2)
+            whenever(it.getChildAt(0)).thenReturn(container1)
+            whenever(it.getChildAt(1)).thenReturn(container2)
+        }
+
+        val sut = fixture.getSut(target)
+
+        sut.onSingleTapUp(event)
+
+        verify(fixture.hub).addBreadcrumb(
+            check<Breadcrumb> {
+                assertEquals("ui.click", it.category)
+                assertEquals("user", it.type)
+                assertEquals("test_button", it.data["view.id"])
+                assertEquals("android.view.View", it.data["view.class"])
+                assertEquals(INFO, it.level)
+            }
+        )
+    }
+
+    @Test
+    fun `ignores invisible or gone views`() {
+        val mockEvent: MotionEvent = mock()
+        val invalidTarget: View = mockView(
+            event = mockEvent,
+            visible = false,
+            clickable = true,
+        )
```

Comment:
This apparently repeats in all the tests, we could have a method with optional params that return the object, it'd reduce to 1 liner most probably